packages/trader/src/AppV2/Hooks/useTradeParamError.ts

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { getDisplayedContractTypes } from 'AppV2/Utils/trade-types-utils';
+
+export type TTradeParams = 'take_profit' | 'stop_loss' | 'date_start';
+
+const useTradeParamError = ({ trade_params }: { trade_params: TTradeParams[] }) => {
+    const { contract_type, proposal_info, validation_errors, trade_type_tab, trade_types } = useTraderStore();
```

Comment:
Instead of relying on validation_errors from trade-store, is it possible to make a proposal request in this hook using useDTraderQuery and handle the logic here? This is to minimize v2's reliance on trade-store gradually.

Reply:
`validation_errors` - is actually a FE validation. Originally it comes from _base-store_ (you can check `validation_errors`, `setValidationErrorMessages` function and `validateProperty` function) and is based on FE validation_rules (an object that contains the validation rules for each property of the store, screenshot is pinned).
The BE validation is coming from `proposal_info` (contains information from `'proposal'` API response).

So, the error is already in proposal_info.
If I add `useDTraderQuery` here, then I'll need to pass the `proposal` request, which is based on current trade store values (that means, that we are still relying on trade store); then I'll receive basically  the same `proposal_info` in response, that we already have here.

Not sure, that right now we can use useDTraderQuery here.

<img width="1550" alt="Screenshot 2024-11-22 at 4 06 45 AM" src="https://github.com/user-attachments/assets/8943ee02-6bd3-4988-b91d-7f3bf8ebf83e">